hsclient/hydroshare.py

```diff
@@ -68,6 +69,33 @@ def url(self):
         return self._file_url
 
 
+def refresh(f):
+    """
+    Decorator for refreshing metadata from HydroShare after the decorated method is called.
+    The docstring of a decorated method is updated to include
+    :param refresh: Defaults True, False to not refresh metadata from HydroShare
+    """
+
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        self = args[0]
+        do_refresh = True
+        if "refresh" in kwargs.keys():
+            do_refresh = kwargs.pop("refresh")
+        result = f(*args, **kwargs)
+        if do_refresh:
+            self.refresh()
+        return result
+
+    # update docstring to include refresh parameter
+    doc_lines = f.__doc__.split("\n")
```

Comment:
That's really helpful to append to the existing docstrings here. Ive used [`python-forge`](https://pypi.org/project/python-forge/) in the past to update method signatures to include additional arguments. It might be worth using here or just using the built-in `inspect` library to update the method signatures with the keyword only argument `refresh`.

Reply:
Thanks for the tip on python-forge, I'll keep it in mind for the future.  Considering this is a minor doc update, I can't just justify pulling in another dependency for it.

Reply:
For sure, that makes sense.